kpi/tasks.py

```diff
@@ -86,52 +82,28 @@ def cleanup_anonymous_exports(**kwargs):
     Task to clean up export tasks created by the AnonymousUser that are older
     than `EXPORT_CLEANUP_GRACE_PERIOD`, excluding those that are still processing
     """
-    BATCH_SIZE = 200
-    cut_off = timezone.now() - timedelta(minutes=config.EXPORT_CLEANUP_GRACE_PERIOD)
-    old_export_ids = (
-        SubmissionExportTask.objects.annotate(
-            processing_seconds=Coalesce(
-                Cast(F('data__processing_time_seconds'), FloatField()), 0.0
-            ),
-            date_modified=ExpressionWrapper(
-                F('date_created') +
-                (F('processing_seconds') * 1000) * timedelta(milliseconds=1),
-                output_field=DateTimeField(),
-            )
-        )
-        .filter(user=get_anonymous_user(), date_modified__lt=cut_off)
-        .exclude(status=ImportExportStatusChoices.PROCESSING)
-        .order_by('pk')
-        .values_list('pk', flat=True)[:BATCH_SIZE]
-    )
+    anon_user = get_anonymous_user()
+    delete_expired_exports(SubmissionExportTask, extra_params={'user': anon_user})
 
-    if not old_export_ids:
-        logging.info('No old anonymous exports to clean up.')
-        return
 
-    deleted_count = 0
-    for export_id in old_export_ids:
-        try:
-            with transaction.atomic():
-                # Acquire a row-level lock without waiting
-                export = (
-                    SubmissionExportTask.objects.only('pk', 'uid', 'result')
-                    .select_for_update(nowait=True)
-                    .get(pk=export_id)
-                )
-
-                if export.result:
-                    try:
-                        export.result.delete(save=False)
-                    except Exception as e:
-                        logging.error(
-                            f'Error deleting file for export {export.uid}: {e}'
-                        )
-                export.delete()
-                deleted_count += 1
-        except DatabaseError:
-            logging.info(f'Export {export_id} is currently being processed. Skipping.')
-    logging.info(f'Cleaned up {deleted_count} old anonymous exports.')
+@celery_app.task
+def cleanup_synchronous_exports(**kwargs):
+    """
+    Task to clean up old synchronous exports that are older than
+    `EXPORT_CLEANUP_GRACE_PERIOD`, excluding those that are still processing
+    """
+    # Do not proceed if grace period is less than cache max age
+    if (
+        config.EXPORT_CLEANUP_GRACE_PERIOD * 60
+        < config.SYNCHRONOUS_EXPORT_CACHE_MAX_AGE
+    ):
+        logging.warning(
+            'Synchronous export cleanup skipped because '
+            'EXPORT_CLEANUP_GRACE_PERIOD is less than '
+            'SYNCHRONOUS_EXPORT_CACHE_MAX_AGE.'
+        )
+        return
+    delete_expired_exports(SubmissionSynchronousExport)
```

Comment:
I'd rather see the validation in Constance. We don't want the exports to stay on the storage if the setting is wrong

Reply:
Too complicated to handle in Constance... Let's just add a warning in Constance.
To be handled by another PR.